### PR TITLE
Added longest word in Javascript

### LINKED ISSUE #3358: Add Longest Word in Javascript
To request a new code snippet, please fill out the following:

Project name: Longest Word
Project link: https://sampleprograms.io/projects/longest-word
Language: Javascript

> The project link must be a link to a project on the [Sample Programs Website project list][sample-programs-project-list].
> If you would like to add a new project to the Sample Programs Website,
> please first make a pull request to the [Sample Programs Website][sample-programs-website].

[sample-programs-website]: https://github.com/TheRenegadeCoder/sample-programs-website
[sample-programs-project-list]: https://sampleprograms.io/projects/

### PR DESCRIPTION
**Pull request by raakesh-r**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][renegade-coder]! 
For simplicity, please make sure that your pull request includes one and only one contribution.

Please fill _one_ of the sections below as applicable.
Please also add any other relevant information to the Notes section at the bottom.
You may delete or just ignore any other sections.
For more information please refer to our [contributing documentation][contributing]

## I Am Adding a New Code Snippet in an Existing Language

- [x] I fixed #3358
- [x] I named the pull request using `Add {PROJECT} in {LANGUAGE}` format

[renegade-coder]: https://therenegadecoder.com/
[contributing-plagiarism]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[contributing-new-project]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#requirements-for-a-new-project
[contributing-new-language]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#requirements-for-a-new-language
[contributing-readme]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#create-readmes
[contributing-tests-in-detail]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#tests-in-detail
[contributing]: ../CONTRIBUTING.md
[sample-programs-project-list]: https://sampleprograms.io/projects/
[contributing-modifications]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#modifying-existing-code-snippets

### CODE REVIEW

File: archive/j/javascript/longest-word.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const error_msg = "Usage: please provide a string";
+const inputString = process.argv[2].trim().replace(/\\n|\\t|\\r/g, ' ')
+const words = inputString.split(' ')
```
Comment (rzuckerm) [inline archive/j/javascript/longest-word.js:3]: These should be done after checking if there are enough command-line arguments.

**Review by rzuckerm — CHANGES_REQUESTED** [review]
Also, it looks like something is going wrong with the longest work calculation. See [these errors](https://github.com/TheRenegadeCoder/sample-programs/actions/runs/6616462305/job/17970794425?pr=3501#step:6:7686)

File: archive/j/javascript/longest-word.js (hunk 1 above)
Comment (rzuckerm) [reply archive/j/javascript/longest-word.js:3]: Also, I'd recommend putting the `trim` after the `replace` to remove leading and trailing spaces.

Comment (raakesh-r) [reply archive/j/javascript/longest-word.js:3]: Thanks @rzuckerm for the suggestions, I have made the changes to the code.

File: archive/j/javascript/longest-word.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const error_msg = "Usage: please provide a string";
+if (process.argv.length != 3) {
+  console.log(error_msg);
+  process.exit(1);
+}
+
+const inputString = process.argv[2].replace(/\\n|\\t|\\r/g, " ");
```
Comment (rzuckerm) [inline archive/j/javascript/longest-word.js:7]: I think this is what is causing the [test failure](https://github.com/TheRenegadeCoder/sample-programs/actions/runs/6617388512/job/17973587051?pr=3501#step:6:5070)
```suggestion
const inputString = process.argv[2].replace(/\n|\t|\r/g, " ");
```

**Review by rzuckerm — CHANGES_REQUESTED** [review]

**Review by rzuckerm — APPROVED** [review]
That change fixed it. Thanks for your contribution.

**Pull request merged**